### PR TITLE
Sitemaps: align with Google deprecations

### LINKED ISSUE #24278: Sitemap: Ensure compatibility with Google changes
### Impacted plugin

Jetpack

### What

Ensure sitemaps are fully compliant with Google

### How

Today, Google announced changes in how they process image and video sitemaps. We should be sure we understand these changes, determine if they make sense to apply (balanced with other sitemap consumers, etc).

https://developers.google.com/search/blog/2022/05/spring-cleaning-sitemap-extensions

We should ensure this lands by Jetpack 11.2 based on the date in the post.

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes #24278

#### Changes proposed in this Pull Request:

* Removes image sitemap fields no longer used.

#### Jetpack product discussion

n/a See https://developers.google.com/search/blog/2022/05/spring-cleaning-sitemap-extensions

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* On a Jetpack connected site with Sitemaps enabled, add images. some images.
* Verify they exist on the image sitemap.
* Run it through Google's sitemap tool ( https://support.google.com/webmasters/answer/7451001 )
* Update to this patch, repeat and confirm it passes still.

### CODE REVIEW

**Review by anomiex — COMMENTED** [review]
Should the code referencing these in sitemap-stylist.php also be cleaned up?
https://github.com/Automattic/jetpack/blob/update/image-sitemap/projects/plugins/jetpack/modules/sitemaps/sitemap-stylist.php#L415-L420

**Comment by bindlegirl** [comment]
@kraftbj should we be moving this to 11.6 milestone?

File: projects/plugins/jetpack/modules/sitemaps/sitemap-stylist.php [hunk 1]
```diff
@@ -412,12 +412,6 @@ public static function image_sitemap_xsl() {
 							<xsl:value-of select='image:image/image:loc'/>
 						</a>
 					</td>
-					<td>
-						<xsl:value-of select='image:image/image:title'/>
-					</td>
-					<td>
-						<xsl:value-of select='image:image/image:caption'/>
-					</td>
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/sitemaps/sitemap-stylist.php:420]: Corresponding `<th>`s need removing on lines 384 and 385.

**Comment by kraftbj** [comment]
Now:
2022-11-08 at 4 48 PM

Before (stable):
2022-11-08 at 4 53 PM

**Review by anomiex — APPROVED** [review]

**Pull request merged**